keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/GaussianVertex.java

```diff
@@ -130,4 +132,12 @@ public DualNumber calculateDualNumber(Map<Vertex, DualNumber> dualNumbers) {
             return DualNumber.createWithRespectToSelf(getId(), getValue());
         }
     }
+
+    @Override
+    public double getLogOfMasterP(List<?> values) {
+        if (values.size() != 1) {
+            throw new IllegalArgumentException("Univariate distributions takes exactly 1 value and you supplied: " + values.size());
+        }
+        return logProb((DoubleTensor) values.get(0));
```

Comment:
You could replace this with a one-liner:

    return logProb((DoubleTensor) Iterables.getOnlyElement(values));

Which gives you a decent exception:

    java.lang.IllegalArgumentException: expected one element but was: <1.0, 1.0>

	at com.google.common.collect.Iterators.getOnlyElement(Iterators.java:322)
	at com.google.common.collect.Iterables.getOnlyElement(Iterables.java:294)
	at io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex.getLogOfMasterP(GaussianVertex.java:140)

Reply:
Nice!